src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -486,46 +496,74 @@ private boolean checkSynchronized(final DetailAST synchronizedAst, boolean useBr
+            lastAST = nextCase.getLastChild();
         }
         else {
-            final int startLineNo = currentCase.getLineNo();
-            for (int i = endLineNo - 2; i > startLineNo - 1; i--) {
-                final int[] line = getLineCodePoints(i);
-                if (!CodePointUtil.isBlank(line)) {
-                    allThroughComment = matchesComment(reliefPattern, i + 1);
-                    break;
-                }
+            ast = currentCase;
+            lastAST = currentCase.getParent().getLastChild();
+        }
+        boolean reliefComment = false;
+        final List<DetailAST> commentContent = getCommentContentNode(ast, lastAST);
+        for (DetailAST detailAST : commentContent) {
+            if (reliefPattern.matcher(detailAST.getText()).find()) {
+                reliefComment = true;
             }
         }
-        return allThroughComment;
+        return reliefComment || commentAsChildOfSameGroup(reliefPattern, currentCase);
     }
 
     /**
-     * Does a regular expression match on the given line and checks that a
-     * possible match is within a comment.
+     * This method will check if the comment are child of same case group ast.
      *
-     * @param pattern The regular expression pattern to use.
-     * @param lineNo The line number in the file.
-     * @return True if a match was found inside a comment.
+     * @param reliefPattern The regular expression pattern to match the comment text
+     * @param currentCase AST of the case that falls through to the next case.
+     * @return true if relief comment found
      */
-    // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/11166
-    @SuppressWarnings("deprecation")
-    private boolean matchesComment(Pattern pattern, int lineNo) {
-        final String line = getLine(lineNo - 1);
-
-        final Matcher matcher = pattern.matcher(line);
-        return matcher.find()
-                && getFileContents().hasIntersectionWithComment(
-                        lineNo, matcher.start(), lineNo, matcher.end());
+    private static boolean commentAsChildOfSameGroup(Pattern reliefPattern,
+                                                     DetailAST currentCase) {
+        boolean reliefComment = false;
+        final List<DetailAST> commentContent = getCommentContentNode(currentCase.getLastChild(),
+                currentCase.getLastChild().getLastChild());
+        for (DetailAST detailAST : commentContent) {
+            if (reliefPattern.matcher(detailAST.getText()).find()) {
+                reliefComment = true;
+            }
+        }
+        return reliefComment;
     }
 
+    /**
+     * This method will return the comment content node between two nodes.
+     *
+     * @param startNode starting node
+     * @param lastNode last node to check
+     * @return list of commentContent node
+     */
+    private static List<DetailAST> getCommentContentNode(DetailAST startNode, DetailAST lastNode) {
```

Comment:
`getCommentContentNodes(fromNode, toNode)`

Change name of method and parameters

Reply:
done